### PR TITLE
Split Organization API and Security API group out of apiextensions

### PR DESCRIPTION
**Pull request by tfussell**
Continuation of https://github.com/giantswarm/giantswarm/issues/19586

This change allows us to update dependencies more freely (no more syncronizing with cluster-api version used by apiextensions). It also aligns this operator with the new structure which is closer to the kubebuilder default.

I will make the PR to apiextensions removing this API there and pulling in the CRD after this is merged.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Review by yulianedyalkova — APPROVED** [review]

File: hack/boilerplate.go.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/*
+Copyright YEAR Giant Swarm GmbH.
```
Comment (marians) [inline hack/boilerplate.go.txt:2]:
```suggestion
Copyright 2022 Giant Swarm GmbH.
```

**Review by marians — APPROVED** [review]
From my shallow perspective, looks good.

File: hack/boilerplate.go.txt (hunk 1 above)
Comment (tfussell) [reply hack/boilerplate.go.txt:2]: `controller-gen` does this for us automatically so we always have the current year :smile:

**Comment by tfussell** [comment]
Tested on `galaxy`. Found an issue with `companyd` missing, but it isn't related and doesn't seem to affect execution.

**Comment by marians** [comment]
As I noticed now, this was missing the CR example. Let's not forget that one in similar future migrations.

**Pull request merged**